2.2/root/opt/app-root/etc/puma.cfg

```diff
@@ -9,6 +9,11 @@ def get_max_memory()
     (2 ** (8*0.size - 2) - 1)
 end
 
+def per_worker_memory()
+    # Unless told otherwise, allow 256MiB per worker
+    ENV.fetch('MEMORY_BYTES_PER_WORKER', '268435456').to_i
```

Comment:
Updated to use `256 * (2**20)`.  I agree that there can't exist any universally correct number here; as Ben says; the issue is that the current mechanism simply doesn't work on any node with a decent CPU, and the knock-on effects are horrible.  Adding MEMORY_BYTES_PER_WORKER as a configurable allows people to override the 256MiB number to allow some form of customised scaling, which was not possible with the previous algorithm.  Please merge.